Library/Homebrew/tap.rb

```diff
@@ -362,6 +362,29 @@ def link_completions_and_manpages
     end
   end
 
+  def fix_remote_configuration
+    return unless remote.include? "github.com"
+
+    current_upstream_head = Utils.popen_read("git", "-C", path, "symbolic-ref", "refs/remotes/origin/HEAD")
+                                 .sub(%r{^refs/remotes/origin/}, "").chomp
+    Utils.popen_read("git", "-C", path, "ls-remote", "--heads", "--exit-code", "origin", current_upstream_head)
+    return if $CHILD_STATUS.success?
+
+    ohai "#{name}: Unable to find the remote branch (#{current_upstream_head})", <<~EOS
+      Searching for a new origin/HEAD...
+    EOS
+
+    repo_data = GitHub.repository(*full_name.split("/"))
+    new_upstream_head = repo_data["default_branch"]
```

Comment:
Isn't this already in `refs/remotes/origin/HEAD`? I'm not sure we need to hit the GitHub API for this.

Reply:
No, `git fetch` doesn't actually change `refs/remotes/origin/HEAD`. That's the key `git` functionality that's missing. It seems that, currently, the only time `refs/remotes/origin/HEAD` is changed is during the clone (when it's written) or manually by `git remote set-head origin <name>`.

I'll double-check, though.

Reply:
I just tested this by editing `.git/refs/remotes/origin/HEAD` to point to a different remote branch. `git remote set-head origin --auto` seemed to do the trick.

Reply:
Hmm, I'm pretty sure there was a reason I chose not to use `git remote set-head origin --auto` but now I'm blanking. It also may have been from a previous attempt...

Is there a reason we would want to avoid running `git fetch origin` rather than `git fetch origin <branch>`?

Using this instead of the GitHub API would I think only require the following commands:

```sh
git fetch origin
git remote set-head origin --auto
# Now the new branch can be found using `git symbolic-ref refs/remotes/origin/HEAD`
git branch -m $old $new
git branch -u origin/$new $new
```

Reply:
`git update-reset` uses `git remote set-head origin --auto` and `git symbolic-ref refs/remotes/origin/HEAD`. See PR https://github.com/Homebrew/brew/pull/7758

> Is there a reason we would want to avoid running git fetch origin rather than git fetch origin <branch>?

`git fetch origin <branch>` can be more efficient and fetch less stuff, depending on how `remote.origin.fetch` in `.git/config` is configured.

Reply:
This would only be run in instances where `git fetch --tags --force origin refs/heads/master:refs/remotes/origin/master` fails, so I think sacrificing some efficiency would be acceptable here. Let me know if that's not a good idea.